src/core_modules/capture-core/components/DataEntries/TeiRegistrationEntry/hooks/useFormValuesFromSearchTerms.js

```diff
@@ -1,19 +1,29 @@
 // @flow
-import { useEffect, useState } from 'react';
+import { useMemo } from 'react';
 import { useSelector } from 'react-redux';
 import { convertClientToForm } from '../../../../converters';
+import type { InputAttribute } from '../../EnrollmentRegistrationEntry/hooks/useFormValues';
 
+type Props = {
+    inheritedAttributes: ?Array<InputAttribute>,
+};
 
-export const useFormValuesFromSearchTerms = () => {
+export const useFormValuesFromSearchTerms = ({ inheritedAttributes }: Props) => {
     const searchTerms = useSelector(({ searchDomain }) => searchDomain.currentSearchInfo.currentSearchTerms);
-    const [formValues, setFormValues] = useState();
-    useEffect(() => {
+
+    return useMemo(() => {
+        if (inheritedAttributes) {
+            return inheritedAttributes
+                ?.reduce((acc, item) => ({
+                    ...acc,
+                    [item.attribute]: convertClientToForm(item.value, item.valueType),
+                }), {});
```

Comment:
I'm not really a fan of using `...acc` when building an object in `reduce`, because it creates an unnecessary copy of `acc` for each element in the array (and the copying gets slower as `acc` gains size). Same thing just below.

Reply:
Same here, I don't think we will reach a point were this code causes performance issues, but you're not wrong either! Refactored to only use the same object 👍